test/unit/DominantJuice.unit.t.sol

```diff
@@ -0,0 +1,962 @@
+        // vm.assume(_notOwner != campaignManager && _notOwner != address(0));
+        successfulCycleHasExpired();
+
+        vm.expectRevert(
+            "AccessControl: account 0x8e24d86be44ab9006bd1277bddc948ecebbfbf6c is missing role 0x5022544358ee0bece556b72ae8983c7f24341bd5b9483ce8a19bff5efbb2de92"
+        );
+        vm.prank(rando);
+        dominantJuice.creatorWithdraw(rando, TOTAL_REFUND_BONUS);
+    }
+
+    function testFuzz_creatorWithdraw_revertsBeforeUnlock(uint256 _seconds) public bonusDeposited {
+        vm.assume(_seconds < cycleExpiryDate + 1209600);
+        vm.warp(_seconds);
+
+        vm.expectRevert("Cycle must be expired and successful, or it must be passed the lock period.");
+        vm.prank(campaignManager);
+        dominantJuice.creatorWithdraw(campaignManager, TOTAL_REFUND_BONUS);
+    }
+
+    function test_creatorWithdraw_revertsIfGoalNotMet() public bonusDeposited {
+        failedCycleHasExpired();
+
+        vm.expectRevert("Cycle must be expired and successful, or it must be passed the lock period.");
+        vm.prank(campaignManager);
+        dominantJuice.creatorWithdraw(campaignManager, TOTAL_REFUND_BONUS);
+    }
+
+    function testFuzz_creatorWithdraw_revertsOnOverdraw(uint256 _amount) public bonusDeposited {
+        successfulCycleHasExpired();
+
+        vm.assume(_amount > dominantJuice.getBalance());
+        vm.expectRevert(DominantJuice.InsufficientFunds.selector);
+        vm.prank(campaignManager);
+        dominantJuice.creatorWithdraw(campaignManager, _amount);
+    }
+
+    function testFuzz_creatorWithdraw_happyPath(uint256 _amount) public bonusDeposited {
+        vm.assume(_amount <= TOTAL_REFUND_BONUS);
+        successfulCycleHasExpired();
+
+        vm.expectEmit(true, true, true, true, address(dominantJuice));
+        emit CreatorWithdrawal(campaignManager, _amount);
+
+        vm.prank(campaignManager);
+        dominantJuice.creatorWithdraw(campaignManager, _amount);
+
+        DominantJuice.FundingStatus memory status = dominantJuice.getCycleFundingStatus();
+
+        if (dominantJuice.getBalance() == 0) {
+            assertEq(true, status.hasCreatorWithdrawnAllFunds);
+        } else {
+            assertEq(false, status.hasCreatorWithdrawnAllFunds);
+        }
+
+        assertEq(TOTAL_REFUND_BONUS - _amount, dominantJuice.getBalance());
+        assertEq(STARTING_BALANCE - TOTAL_REFUND_BONUS + _amount, campaignManager.balance);
+    }
+
+    function test_creatorWithdraw_sendsFundsToDifferentAddress() public bonusDeposited {
+        successfulCycleHasExpired();
+
+        vm.prank(campaignManager);
+        dominantJuice.creatorWithdraw(rando, TOTAL_REFUND_BONUS);
+
+        assertEq(0, dominantJuice.getBalance());
+        assertEq(rando.balance, STARTING_BALANCE + TOTAL_REFUND_BONUS);
+    }
+
+    function testFuzz_creatorWithdraw_creatorCanCallAfterUnlock(uint256 _seconds) public bonusDeposited {
+        successfulCycleHasExpired();
```

Comment:
though this is true, can't they also withdraw on failed cycle has expired?

Reply:
Haha, dangit. Thanks. Yes, will change that `failedCycleHasExpired`.